recipes/ruby/all/conanfile.py

```diff
@@ -122,75 +178,97 @@ def generate(self):
+                copy(self, '*.a', src=dirname, dst=os.path.join(self.package_folder, dst), keep_path=True)
+                copy(self, '*.lib', src=dirname, dst=os.path.join(self.package_folder, dst), keep_path=True)
 
     def package_info(self):
         binpath = os.path.join(self.package_folder, "bin")
         self.output.info(f"Adding to PATH: {binpath}")
         self.env_info.PATH.append(binpath)
 
-        version = tools.Version(self.version)
-        rubylib = self.cpp_info.components["rubylib"]
+        version = Version(self.version)
         config_file = glob.glob(os.path.join(self.package_folder, "include", "**", "ruby", "config.h"), recursive=True)[0]
-        rubylib.includedirs = [
+        self.cpp_info.includedirs = [
             os.path.join(self.package_folder, "include", f"ruby-{version}"),
             os.path.dirname(os.path.dirname(config_file))
         ]
-        rubylib.libs = tools.collect_libs(self)
+        self.cpp_info.libs = collect_libs(self)
         if is_msvc(self):
             if self.options.shared:
-                rubylib.libs = list(filter(lambda l: not l.endswith("-static"), rubylib.libs))
+                self.cpp_info.libs = list(filter(lambda l: not l.endswith("-static"), self.cpp_info.libs))
             else:
-                rubylib.libs = list(filter(lambda l: l.endswith("-static"), rubylib.libs))
-        rubylib.requires.extend(["zlib::zlib", "gmp::gmp"])
-        if self.options.with_openssl:
-            rubylib.requires.append("openssl::openssl")
+                self.cpp_info.libs = list(filter(lambda l: l.endswith("-static"), self.cpp_info.libs))
+
         if self.settings.os in ("FreeBSD", "Linux"):
-            rubylib.system_libs = ["dl", "pthread", "rt", "m", "crypt"]
+            self.cpp_info.system_libs = ["dl", "pthread", "rt", "m", "crypt", "util"]
         elif self.settings.os == "Windows":
-            rubylib.system_libs = self._windows_system_libs
+            self.cpp_info.system_libs = self._windows_system_libs
         if str(self.settings.compiler) in ("clang", "apple-clang"):
-            rubylib.cflags = ["-fdeclspec"]
-            rubylib.cxxflags = ["-fdeclspec"]
-        if tools.is_apple_os(self.settings.os):
-            rubylib.frameworks = ["CoreFoundation"]
+            self.cpp_info.cflags = ["-fdeclspec"]
+            self.cpp_info.cxxflags = ["-fdeclspec"]
+        if is_apple_os(self):
+            self.cpp_info.frameworks = ["CoreFoundation"]
 
-        self.cpp_info.filenames["cmake_find_package"] = "Ruby"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "Ruby"
+        self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_file_name", "Ruby")
+        self.cpp_info.set_property("cmake_target_name", "Ruby::Ruby")
+        self.cpp_info.set_property("pkg_config_name", "ruby")
+        self.cpp_info.set_property("pkg_config_aliases", [f"ruby-{version.major}.{version.minor}"])
```

Comment:
I would move it to the top of `package_info()` as it's the most important bit of info to the package consumers, really, and it's kind of the conventional location in other recipes.